src/metabase/models/parameter_card.clj

```diff
@@ -45,27 +47,30 @@
   ([parameterized-object-type parameterized-object-id parameter-ids-still-in-use]
    (db/delete! ParameterCard
                :parameterized_object_type parameterized-object-type
-               :parameterized_object_id  parameterized-object-id
+               :parameterized_object_id parameterized-object-id
                (if (empty? parameter-ids-still-in-use)
-                {}
-                {:where [:not-in :parameter_id parameter-ids-still-in-use]}))))
+                 {}
+                 {:where [:not-in :parameter_id parameter-ids-still-in-use]}))))
 
-(defn- upsert-for-dashboard!
-  [dashboard-id parameters]
+(defn- upsert-from-parameters!
+  [parameterized-object-type parameterized-object-id parameters]
   (doseq [{:keys [values_source_config id]} parameters]
     (let [card-id    (:card_id values_source_config)
-          conditions {:parameterized_object_id   dashboard-id
-                      :parameterized_object_type "dashboard"
+          conditions {:parameterized_object_id   parameterized-object-id
+                      :parameterized_object_type parameterized-object-type
                       :parameter_id              id}]
       (or (db/update-where! ParameterCard conditions :card_id card-id)
           (db/insert! ParameterCard (merge conditions {:card_id card-id}))))))
 
-(defn upsert-or-delete-for-dashboard!
-  "Create, update, or delete appropriate ParameterCards for each parameter in the dashboard"
-  [{dashboard-id :id parameters :parameters}]
+(s/defn upsert-or-delete-from-parameters!
+  "From a parameters list on card or dashboard, create, update,
+  or delete appropriate ParameterCards for each parameter in the dashboard"
+  [parameterized-object-type :- su/NonBlankString
```

Comment:
should we use the `valid-parameterized_object_type` set to lock down this schema properly?